### PR TITLE
Homepage controller porting

### PR DESCRIPTION
**Pull request by antodippo**
Ported the HomeController from the old versione, with tests and old Twig template.

### CODE REVIEW

**Comment by JellyBellyDev** [comment]
Please squash your commit and rebase with last version of PUGX:release/v3.0.0!
Thanks!

File: config/services.yaml [hunk 1]
```diff
@@ -18,6 +18,9 @@ parameters:
         - { name: 'composerlock', label: 'composer.lock', route: "pugx_badge_composerlock" }
         - { name: 'gitattributes', label: '.gitattributes', route: "pugx_badge_gitattributes" }
 
+    redis.reader.class: 'Snc\RedisBundle\DependencyInjection\SncRedisExtension'
+    pugx.poser.class: 'PUGX\Poser\Poser'
```
Comment (JellyBellyDev) [inline config/services.yaml]: Sorry, why do you define these services? You can not use the autowire?

File: config/services.yaml [hunk 2]
```diff
@@ -53,6 +58,25 @@ services:
...
+
+    PUGX\Poser\SvgPlasticRender:
+        class: PUGX\Poser\Render\SvgRender
+
+    PUGX\Poser\SvgFlatRender:
+        class: PUGX\Poser\Render\SvgFlatRender
+
+    PUGX\Poser\SvgFlatSquareRender:
+        class: PUGX\Poser\Render\SvgFlatSquareRender
+
```
Comment (JellyBellyDev) [inline config/services.yaml]: Sorry, would you change these definitions alphabetically?
thank you

File: src/Controller/HomeController.php [hunk 3]
```diff
@@ -2,27 +2,58 @@
 
 namespace App\Controller;
 
-use App\Contributors\Service\Repository as ContributorsRepository;
-use App\Contributors\Model\Contributor;
```
Comment (JellyBellyDev) [inline src/Controller/HomeController.php]: Why you remove logic for get contributors?

File: src/Controller/HomeController.php [hunk 4]
```diff
@@ -2,27 +2,58 @@
...
+    /**
+     * HomeController constructor.
+     * @param ReaderInterface $statsReader
+     * @param Poser $poser
+     */
+    public function __construct(ReaderInterface $statsReader, Poser $poser)
+    {
+        $this->statsReader = $statsReader;
+        $this->poser = $poser;
+    }
```
Comment (JellyBellyDev) [inline src/Controller/HomeController.php]: if you use these services only in `indexAction` would not it be better to inject them only to the `indexAction`?

File: src/Service/ReaderInterface.php [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace App\Service;
+
+/**
+ * Interface ReaderInterface
+ * @package App\Service
+ *
+ * @author Roberto Umbertini <daeronmalnwe@gmail.com>
+ */
+interface ReaderInterface
```
Comment (JellyBellyDev) [inline src/Service/ReaderInterface.php]: This interface is not the same `Stats/Reader/ReaderInterface.php` by @garak?

File: src/Service/RedisReader.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+use Predis\Client;
+
+/**
+ * Class RedisReader
+ * @package App\Service
+ *
+ * @author Roberto Umbertini <daeronmalnwe@gmail.com>
+ */
+class RedisReader implements ReaderInterface
```
Comment (JellyBellyDev) [inline src/Service/RedisReader.php]: also this class is yet present here `Stats/Reader/RedisReader.php` by @garak!

File: src/Service/ReaderInterface.php (hunk 5 above)
Comment (antodippo) [reply src/Service/ReaderInterface.php]: Probably @daeronmalnwe did it at the same time... Rob can you take a look?

File: src/Service/RedisReader.php (hunk 6 above)
Comment (antodippo) [reply src/Service/RedisReader.php]: @daeronmalnwe again, can you take a look? :)

File: config/services.yaml (hunk 1 above)
Comment (antodippo) [reply config/services.yaml]: `redis.reader.class` probably will go away with @daeronmalnwe review, while `pugx.poser.class` is for testing purpose, in `services_test.yml` I've specified a PoserStub in it

File: src/Controller/HomeController.php (hunk 3 above)
Comment (antodippo) [reply src/Controller/HomeController.php]: Probably a bad conflict resolution, I'll check

**Comment by JellyBellyDev** [comment]
sorry but I close in favor of this: https://github.com/PUGX/badge-poser/pull/219

**Pull request closed without merging**